### DISCUSSION TITLE
Asciidoctor integration

### DISCUSSION THREAD

**Discussion by AmurgCodru** [nvim-telekasten/telekasten.nvim#216] [Ideas]
Hi

I think it would make tons of sense to add support for asciidoctor.

I've used many note taking apps, even started working on an asciidoctor one. While markdown may seem easy, it has too many limitations and adding anything else to it is hackish and may not be compatible with anything else, thus archiving wise it's very sad. Asciidoctor would provide a full set of features similar to markdown. Yet it provides a plethora of amazing extra's

- Charts/Graphs directly from asciidoctor
- It produces beautiful output in HTML by default 
- you can also generate PDF's AND  epub's
-  footer/header integrations
- footnotes accumulation
- code highlighting in output html/pdf/epub 
- Multi page/files includes -
- Native header support for settings/variables 
- Variable support out of the box
- Admonition, sidebars, block titles
- Generate presentations out of the box

I could list them all, the purpose is that it helps enhance productivity and thus makes a note taking app much more flexible and extensible to many other problems which might need to be solved. 

Most of them are amazing if you think about writing a book without getting out of asciidoctor and "hacking" away and adding complexity to code.

**Comment by lambtho12** [comment]
Adding support for a whole other language would be extremely challenging and time-consuming. 
I am already unhappy with the "messy" state of the codebase as it is now with only markdown support. It would likely be a nightmare to maintain if we were to add asciidoc.

I understand the benefits an individual can find in using asciidoc, but I think that markdown is way more ubiquitous (thanks to its simplicity). It would make very little sense to increase the complexity of the plugin 10x for something that is only used by a fraction of the userbase.
Obviously, if there is a serious push for asciidoc, this is something we might reconsider at some point. Once telekasten is properly refactored, the functions are cleaned up and the syntax and stuff are isolted, this is something we could revisit. But for now, I'd rather the very little time I can on fixing bugs, adding basic features or moving on with the refactoring.

As for my personal opinion (mileage may vary here), I use telekasten to write notes (personal or work related), not complete software documentation. This is how it was originally meant to be used (based on the zettelkasten principles). Whenever I want to write a complete documentation with may complex features, I move to latex or some other heavy-lifting language.
